### PR TITLE
FI-964 fix bugs with _id search parameter and .where in search validation

### PR DESCRIPTION
**Pull request by czh-orz**
This PR addresses some bugs in the generic generator involving search parameters. One issue is that the search validation code used to have fhir path expressions that we couldn't handle. For example, the us core laboratory result sequence had the ".where" in the search validation code. This removes the ".where" part. The second is that the patient id parameter was being set to `id` instead of `_id`.  

To test, run the generic generator on the us core resources. Check that the patient sequence's id search is _id instead id and that the ".where" is not included in the lab result sequence.

### CODE REVIEW

File: generator/search_parameter_metadata.rb [hunk 1]
```diff
@@ -36,7 +36,18 @@ def type
...
+      end
+
+      def expression_without_fhir_path(path)
+        # handle some fhir path stuff. Remove this once fhir path server is added
+        as_type = path.scan(/.as\((.*?)\)/).flatten.first
+        path = path.gsub(/.as\((.*?)\)/, capitalize_first_letter(as_type)) if as_type.present?
+        path.gsub(/.where\((.*)/, '')
+      end
+
+      def capitalize_first_letter(str)
```
[the remark this answers is not in the document]
Comment (okeefm) [reply generator/search_parameter_metadata.rb:49]: Man, ActiveSupport has everything, doesn't it?

[the remark this answers is not in the document]
Comment (czh-orz) [reply generator/search_parameter_metadata.rb]: 👍

**Review by radamson — APPROVED** [review]
I tested these bug fixes for the tech forum demo and while generating some mcode tests.

**Review by okeefm — APPROVED** [review]
Looks good. Will merge when Travis finishes.

**Pull request merged**